src/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -350,10 +353,17 @@ protected WgsMetrics generateWgsMetrics() {
     }
 
     protected long getBasesExcludedBy(final CountingFilter filter) {
+        if(INTERVALS != null){
+            return filter.getFilteredRecords();
+        }
         return filter.getFilteredBases();
     }
 
     protected SamLocusIterator getLocusIterator(final SamReader in) {
+        if(INTERVALS != null){
+            IOUtil.assertFileIsReadable(INTERVALS);
```

Comment:
Could you add this check to the start of the `doWork` method? I think the pattern is to call the `IOUtil` asserts as soon as possible directly in the `doWork` the method.